src/Adapter/Product/CustomizationFieldDeleter.php

```diff
@@ -0,0 +1,141 @@
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Exception\CannotDeleteCustomizationFieldException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\Exception\CustomizationFieldException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Customization\ValueObject\CustomizationFieldId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotUpdateProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductException;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShopException;
+
+/**
+ * Deletes customization field/fields using legacy object models
+ */
+final class CustomizationFieldDeleter implements CustomizationFieldDeleterInterface
+{
+    /**
+     * @var CustomizationFieldProvider
+     */
+    private $customizationFieldProvider;
+
+    /**
+     * @var ProductProvider
+     */
+    private $productProvider;
+
+    /**
+     * @var ProductUpdater
+     */
+    private $productUpdater;
+
+    /**
+     * @param CustomizationFieldProvider $customizationFieldProvider
+     * @param ProductProvider $productProvider
+     * @param ProductUpdater $productUpdater
+     */
+    public function __construct(
+        CustomizationFieldProvider $customizationFieldProvider,
+        ProductProvider $productProvider,
+        ProductUpdater $productUpdater
+    ) {
+        $this->customizationFieldProvider = $customizationFieldProvider;
+        $this->productProvider = $productProvider;
+        $this->productUpdater = $productUpdater;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(CustomizationFieldId $customizationFieldId): void
+    {
+        $customizationField = $this->customizationFieldProvider->get($customizationFieldId);
+        $this->performDeletion($customizationField, CannotDeleteCustomizationFieldException::FAILED_DELETE);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bulkDelete(array $customizationFieldIds): void
+    {
+        foreach ($customizationFieldIds as $customizationFieldId) {
+            $customizationField = $this->customizationFieldProvider->get($customizationFieldId);
+            $this->performDeletion($customizationField, CannotDeleteCustomizationFieldException::FAILED_BULK_DELETE);
+        }
+    }
+
+    /**
+     * @param CustomizationField $customizationField
+     * @param int $errorCode
+     *
+     * @return void
+     *
+     * @throws CannotDeleteCustomizationFieldException
+     * @throws CustomizationFieldException
+     * @throws ProductException
+     * @throws ProductNotFoundException
+     */
+    private function performDeletion(CustomizationField $customizationField, int $errorCode): void
+    {
+        $product = $this->productProvider->get(new ProductId((int) $customizationField->id_product));
```

Comment:
I'm afraid this is not very optimized, most of the time all customization fields have the same product, so we could fetch it only once.

Not of course nothing ensures that they do have the same product, but then maybe we could add a private `getProduct` method that would store the product by ID thus avoiding creating new instances of `ProductId` and `Product` for each deleted customization field